### ISSUE TITLE
Question about implementation detailes

### ISSUE THREAD

**Issue by WangHanLinHenry** [yuyq18/StepTool#11]
Thanks for your excellent work! 

I have two questions to ask.

1. When I tried to directly run the command "bash scripts/steptool_train/train_toolllama.sh" and train on the "step_grained_for_example.csv" file, i found there happens some **"Nan" value** in some metrics at the initial epochs which are shown followings. Have you meet this kind of phenomena and how to solve this?
2. How long it takes to conduct sft training and ppo training?

Looking forward to your reply!

Epoch 0:
stats: {'objective/kl': array(0., dtype=float32), 'objective/kl_dist': array([0., 0., 0., 0., 0., 0., 0., 0.], dtype=float32), 'objective/logprobs': array([[-10.373491, -10.373491, -10.373491, ..., -10.373491, -10.373491,
        -10.373491],
       [-10.373491, -10.373491, -10.373491, ..., -10.373491, -10.373491,
        -10.373491],
       [-10.373491, -10.373491, -10.373491, ..., -10.373491, -10.373491,
        -10.373491],
       ...,
       [-10.373491, -10.373491, -10.373491, ..., -10.373491, -10.373491,
        -10.373491],
       [-10.373491, -10.373491, -10.373491, ..., -10.373491, -10.373491,
        -10.373491],
       [-10.373491, -10.373491, -10.373491, ..., -10.373491, -10.373491,
        -10.373491]], dtype=float32), 'objective/ref_logprobs': array([[-10.373491, -10.373491, -10.373491, ..., -10.373491, -10.373491,
        -10.373491],
       [-10.373491, -10.373491, -10.373491, ..., -10.373491, -10.373491,
        -10.373491],
       [-10.373491, -10.373491, -10.373491, ..., -10.373491, -10.373491,
        -10.373491],
       ...,
       [-10.373491, -10.373491, -10.373491, ..., -10.373491, -10.373491,
        -10.373491],
       [-10.373491, -10.373491, -10.373491, ..., -10.373491, -10.373491,
        -10.373491],
       [-10.373491, -10.373491, -10.373491, ..., -10.373491, -10.373491,
        -10.373491]], dtype=float32), 'objective/kl_coef': 0.3, 'objective/entropy': array(2309.3984, dtype=float32), 'ppo/mean_non_score_reward': array(0., dtype=float32), 'ppo/mean_scores': array(0.5, dtype=float32), 'ppo/std_scores': array(0.5345225, dtype=float32), 'tokens/queries_len_mean': 686.5, 'tokens/queries_len_std': 247.3701629638672, 'tokens/queries_dist': array([655., 939., 383., 897., 302., 661., 688., 967.], dtype=float32), 'tokens/responses_len_mean': 126.625, 'tokens/responses_len_std': 38.47053527832031, 'tokens/responses_dist': array([108., 110.,  97., 145., 132., 106., 102., 213.], dtype=float32), 'ppo/loss/policy': array([nan], dtype=float32), 'ppo/loss/value': array([nan], dtype=float32), 'ppo/loss/total': array([nan], dtype=float32), 'ppo/policy/entropy': array([nan], dtype=float32), 'ppo/policy/approxkl': array([nan], dtype=float32), 'ppo/policy/policykl': array([nan], dtype=float32), 'ppo/policy/clipfrac': array([0.], dtype=float32), 'ppo/policy/advantages': array([-1., -1., -1., ..., -1., -1., -1.], dtype=float32), 'ppo/policy/advantages_mean': array([nan], dtype=float32), 'ppo/policy/ratio': array([ 1.,  1.,  1., ..., nan, nan, nan], dtype=float32), 'ppo/returns/mean': array([nan], dtype=float32), 'ppo/returns/var': array([nan], dtype=float32), 'ppo/val/vpred': array([nan], dtype=float32), 'ppo/val/error': array([nan], dtype=float32), 'ppo/val/clipfrac': array([0.00013298], dtype=float32), 'ppo/val/mean': array([nan], dtype=float32), 'ppo/val/var': array([nan], dtype=float32), 'ppo/policy/advantages_batch': array([nan, nan, nan, nan, nan, nan, nan, nan], dtype=float32), 'ppo/val/var_explained': array([nan], dtype=float32), 'ppo/learning_rate': 1e-05, 'time/ppo/forward_pass': 65.88533544540405, 'time/ppo/compute_rewards': 0.026012659072875977, 'time/ppo/compute_advantages': 0.07614898681640625, 'time/ppo/optimize_step': 148.2997899055481, 'time/ppo/calc_stats': 1.0381104946136475, 'time/ppo/total': 215.3256652355194}

Epoch 1:
stats: {'objective/kl': array(nan, dtype=float32), 'objective/kl_dist': array([nan, nan, nan, nan, nan, nan, nan, nan], dtype=float32), 'objective/logprobs': array([[nan, nan, nan, ..., nan, nan, nan],
       [nan, nan, nan, ..., nan, nan, nan],
       [nan, nan, nan, ..., nan, nan, nan],
       ...,
       [nan, nan, nan, ..., nan, nan, nan],
       [nan, nan, nan, ..., nan, nan, nan],
       [nan, nan, nan, ..., nan, nan, nan]], dtype=float32), 'objective/ref_logprobs': array([[nan, nan, nan, ..., nan, nan, nan],
       [nan, nan, nan, ..., nan, nan, nan],
       [nan, nan, nan, ..., nan, nan, nan],
       ...,
       [nan, nan, nan, ..., nan, nan, nan],
       [nan, nan, nan, ..., nan, nan, nan],
       [nan, nan, nan, ..., nan, nan, nan]], dtype=float32), 'objective/kl_coef': 0.299952, 'objective/entropy': array(nan, dtype=float32), 'ppo/mean_non_score_reward': array(nan, dtype=float32), 'ppo/mean_scores': array(0.75, dtype=float32), 'ppo/std_scores': array(0.46291006, dtype=float32), 'tokens/queries_len_mean': 624.125, 'tokens/queries_len_std': 243.69444274902344, 'tokens/queries_dist': array([702., 688., 897., 302., 383., 967., 399., 655.], dtype=float32), 'tokens/responses_len_mean': 133.75, 'tokens/responses_len_std': 41.59584426879883, 'tokens/responses_dist': array([173., 102., 145., 132.,  97., 213., 100., 108.], dtype=float32), 'ppo/loss/policy': array([nan], dtype=float32), 'ppo/loss/value': array([nan], dtype=float32), 'ppo/loss/total': array([nan], dtype=float32), 'ppo/policy/entropy': array([nan], dtype=float32), 'ppo/policy/approxkl': array([nan], dtype=float32), 'ppo/policy/policykl': array([nan], dtype=float32), 'ppo/policy/clipfrac': array([0.], dtype=float32), 'ppo/policy/advantages': array([-1., -1., -1., ..., -1., -1., -1.], dtype=float32), 'ppo/policy/advantages_mean': array([nan], dtype=float32), 'ppo/policy/ratio': array([nan, nan, nan, ..., nan, nan, nan], dtype=float32), 'ppo/returns/mean': array([nan], dtype=float32), 'ppo/returns/var': array([nan], dtype=float32), 'ppo/val/vpred': array([nan], dtype=float32), 'ppo/val/error': array([nan], dtype=float32), 'ppo/val/clipfrac': array([0.], dtype=float32), 'ppo/val/mean': array([nan], dtype=float32), 'ppo/val/var': array([nan], dtype=float32), 'ppo/policy/advantages_batch': array([nan, nan, nan, nan, nan, nan, nan, nan], dtype=float32), 'ppo/val/var_explained': array([nan], dtype=float32), 'ppo/learning_rate': 1e-05, 'time/ppo/forward_pass': 65.41276669502258, 'time/ppo/compute_rewards': 0.007170200347900391, 'time/ppo/compute_advantages': 0.06346702575683594, 'time/ppo/optimize_step': 148.22121691703796, 'time/ppo/calc_stats': 0.9623098373413086, 'time/ppo/total': 214.66714453697205}
1it [03:34, 214.67s/it]
